### PR TITLE
issue #3497 : Removed deprecated enabled flag from InMem telemetry config

### LINKED ISSUE #3497: Remove deprecated "enabled" flag from InMem telemetry config
The "enabled" flag was deprecated as part of #3492, which changed the default value to false for the 1.5.0 release. We warn if the flag is explicitly set since it is no longer needed with the changed default.

We can remove the deprecated flag in 1.6.0.

### PR DESCRIPTION
**Pull request by vish-k**
…nfig

**Pull Request check list**

- [ x] Commit conforms to CONTRIBUTING.md?
- [ x] Proper tests/regressions included?
- [ x] Documentation updated?

**Affected functionality**

**Description of change**

**Which issue this PR fixes**
`fixes #3497` Remove deprecated "enabled" flag from InMem telemetry config

### CODE REVIEW

**Review by azdagron — APPROVED** [review]
Thanks, @vish-k !

**Comment by azdagron** [comment]
Looks like we need DCO signoff on your commit.l You can follow the details of the failed DCO workflow for a description of how to solve.

**Comment by vish-k** [comment]
Thanks for your patience. I followed the steps to add DCO commit. Pls let me know if this works.

**Comment by azdagron** [comment]
Looks like it is still missing. I'd rebase the commit on origin/main and then do a `git commit -s --amend` to add the signoff, and then force push to the PR branch.

**Comment by azdagron** [comment]
DCO signoff is still unhappy.

**Comment by vish-k** [comment]
It's still complaining about earlier commit [17d7827] which I deleted. The latest commit has "Signed-off-by". Is there any way to bypass check for deleted commit? I'm sorry, this is my first commit so I'm messing this up. Should I just redo the fork and create new PR?

**Comment by azdagron** [comment]
That commit is still part of the PR branch. I suggest rebasing your PR branch on top of main instead of merging main.

**Comment by vish-k** [comment]
Since I was working a fork in my repo, I didn't create separate branch. I just made changes to local main branch, pushed to main in my repo/main and PR is merge my main into spire/main.

**Comment by azdagron** [comment]
Sure, that's fine. So the main branch on your fork is the PR branch. I'd suggest rebasing main on your fork on top of main from this repo.

**Comment by azdagron** [comment]
Unfortunately DCO is not something we can skip. If you hit me up on the SPIFFE slack, I'd be happy to help you get this sorted.

**Comment by azdagron** [comment]
Looks like the linter is failing:
```
pkg/common/telemetry/inmem_test.go:13: unnecessary leading newline (whitespace)
```

**Comment by vish-k** [comment]
Ok. I will check it.

**Comment by vish-k** [comment]
It's failing with "unnecessary leading newline (whitespace)" but there is no whitespace or newline. 

I tried vs code, gofmt and even nolint directive for golangci-lint but no luck so far. Do you have any suggestion?

File: pkg/common/telemetry/inmem_test.go [hunk 1]
```diff
@@ -11,8 +11,6 @@ import (
 )
 
 func TestInMem(t *testing.T) {
-	enabled := true
-	disabled := false
```
Comment (azdagron) [inline pkg/common/telemetry/inmem_test.go:14]: Remove this empty line at the beginning of the TestInMem function to satisfy the linter.

Comment (vish-k) [reply pkg/common/telemetry/inmem_test.go]: Thanks! fixed it.

**Review by azdagron — APPROVED** [review]

**Pull request merged**